### PR TITLE
[EVM] Update EVM contract for Cadence 1.0

### PR DESCRIPTION
**Pull request by sisyphusSmiling**
Closes: #5350

Adds view modifiers and entitlements to the EVM Cadence contract

### CODE REVIEW

File: fvm/evm/stdlib/contract.cdc [hunk 1]
```diff
@@ -102,14 +109,20 @@ contract EVM {
...
+        view fun address(): EVMAddress {
             // Always create a new EVMAddress instance
             return EVMAddress(bytes: self.addressBytes)
         }
 
+        /// The EVM address of the cadence owned account behind an entitlement, acting as proof of access
+        access(Validate)
+        view protectedAddress(): EVMAddress {
+            return self.address()
+        }
```
Comment (turbolent) [inline fvm/evm/stdlib/contract.cdc:121]: What is this for?

Comment (sisyphusSmiling) [reply fvm/evm/stdlib/contract.cdc:121]: My thinking here is that it serves as proof that the owner of this reference does in fact own this COA. Consider something like

```cadence
access(all) fun claimAirdrop(coa: auth(EVM.Validate) &EVM.CadenceOwnedAccount): @ExampleToken.Vault? {
  if let dropAmount = self.allowlist[coa.address()] {
    return <- self.mintTokens(amount: dropAmount)
  }
  return nil
}
```

Without something like the above, how would one prove ownership of a COA without also exposing sensitive methods like `call`, `deploy`, etc or necessitating a signature?

Comment (turbolent) [reply fvm/evm/stdlib/contract.cdc:121]: I see 👍

File: fvm/evm/stdlib/contract.cdc [hunk 2]
```diff
@@ -4,8 +4,15 @@ import "FlowToken"
 access(all)
 contract EVM {
 
-     access(all)
-     event CadenceOwnedAccountCreated(addressBytes: [UInt8; 20])
+    // Entitlements enabling finer-graned access control on a CadenceOwnedAccount
+    access(all) entitlement Validate
+    access(all) entitlement Withdraw
```
Comment (ramtinms) [inline fvm/evm/stdlib/contract.cdc:9]: I don't know much about entitlements but do we need to also update the validateCOAOwnershipProof to use a Validate entitlement?

Comment (turbolent) [reply fvm/evm/stdlib/contract.cdc:9]: That isn't needed, and would actually make the function inaccessible: `validateCOAOwnershipProof` is defined on the contract (`EVM`), not on the resource (`CadenceOwnedAccount`)

Comment (ramtinms) [reply fvm/evm/stdlib/contract.cdc:9]: got it.

**Review by ramtinms — APPROVED** [review]

**Comment by turbolent** [comment]
@sisyphusSmiling Can you please propose to update the FLIP with these changes?

**Comment by sisyphusSmiling** [comment]
> @sisyphusSmiling Can you please propose to update the FLIP with these changes?

Added a PR against the FLIP branch - https://github.com/onflow/flips/pull/249

**Review by janezpodhostnik — APPROVED** [review]

**Comment by turbolent** [comment]
Tests fail, because `InternalEVM`'s functions themself are not `view`. I'll push a fix

**Comment by turbolent** [comment]
Should be fixed by [175ef4a](https://github.com/onflow/flow-go/pull/5390/commits/175ef4abd6243ac8f29ce89c1ec2b53aaf2b0d7a) and [7cbb42e](https://github.com/onflow/flow-go/pull/5390/commits/7cbb42ec5cac3c240db0f8aeaae53864e40c3d2a)

**Review by turbolent — APPROVED** [review]

**Pull request merged**